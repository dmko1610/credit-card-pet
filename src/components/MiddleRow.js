import React from "react";
import PropTypes from "prop-types";
import { View, Text, StyleSheet } from "react-native";
import { useSelector } from "react-redux";

const BankIdentificationNumber = ({ bin }) => {
  return <Text style={styles.number}>{bin}</Text>;
};

/* const AccountIdentifierNumber = ({}) => {
  return <Text>78 9876 5432</Text>;
}; */

const MiddleRow = () => {
  const cardNumber = useSelector((state) => state.root.cardNumber);
  const isNumberFocused = useSelector((state) => state.focus.isNumberFocused);
  const containerStyle = [styles.middleRowContainer];
  if (isNumberFocused) {
    containerStyle.push(styles.focusedContainer);
  }
  return (
    <View style={StyleSheet.flatten(containerStyle)}>
      <BankIdentificationNumber bin={cardNumber} />
      {/* <AccountIdentifierNumber style={number} /> */}
    </View>
  );
};

const styles = StyleSheet.create({
  middleRowContainer: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    marginHorizontal: 10,
  },
  focusedContainer: {
    borderWidth: 2,
    borderColor: "#FFFFFF",
    borderRadius: 10,
  },
  number: {
    letterSpacing: 2,
    color: "black",
    fontSize: 25,
    fontFamily: "monospace",
  },
});

BankIdentificationNumber.propTypes = {
  bin: PropTypes.string,
};

export default MiddleRow;
