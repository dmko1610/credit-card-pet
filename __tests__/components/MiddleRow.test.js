import React from "react";
import renderer from "react-test-renderer";

import MiddleRow from "../../src/components/MiddleRow";

describe("Middle Row Creation", () => {
  it("Render component", () => {
    const tree = renderer.create(<MiddleRow />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
