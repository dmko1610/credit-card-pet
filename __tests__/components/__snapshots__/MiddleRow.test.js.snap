// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Middle Row Creation Render component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "monospace",
        "fontSize": 25,
        "letterSpacing": 2,
      }
    }
  >
    1234 56
  </Text>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "monospace",
        "fontSize": 25,
        "letterSpacing": 2,
      }
    }
  >
    78 9876 5432
  </Text>
</View>
`;
